localstack/aws/connect.py

```diff
@@ -396,16 +399,26 @@ def get_client(
         :param config: Boto config for advanced use.
         """
 
+        if config is None:
+            config = self._config
+        else:
+            config = self._config.merge(config)
+
+        endpoint_url = endpoint_url or get_local_service_url(service_name)
+        if service_name == "s3":
+            if re.match(r"https?://localhost(:[0-9]+)?", endpoint_url):
+                endpoint_url = endpoint_url.replace("://localhost", "://%s" % get_s3_hostname())
```

Comment:
Why not use an fstring here?

Reply:
copypasta :spaghetti: 

I'll update it though :+1: